eos/utils/kinematic.hh

```diff
@@ -223,6 +233,68 @@ namespace eos
             ///@}
     };
 
+    /*!
+     * Base class for all users of Kinematics objects.
+     */
+    class KinematicUser
+    {
+        protected:
+            std::set<KinematicVariable::Id> _ids;
+
+        public:
+            ///@name Iteration over ids
+            ///@{
+            struct ConstIteratorTag;
+            using ConstIterator = WrappedForwardIterator<ConstIteratorTag, const KinematicVariable::Id>;
+
+            ConstIterator begin_kinematics() const;
+            ConstIterator end_kinematics() const;
+            ///@}
+
+            ///@name Access
+            ///@{
+            /*!
+             * Remove a kinematics variable from the set of used ids.
+             */
+            void drop(const KinematicVariable::Id & id);
+
+            /*!
+             * Add a given kinematics variable id to our list of used ids.
+             *
+             * @param id   The kinematics variable id that we use.
+             */
+            void uses_kinematic(const KinematicVariable::Id & id);
```

Comment:
I think this should be ``uses`` just like for ``ParameterUser``. We can and should select which base class we mean by recasting, right?
```C++
reinterpret_cast<ParameterUser *>(this)->uses(...);
reinterpret_cast<KinematicUser *>(this)->uses(...);
```